### DISCUSSION TITLE
Can't use useLightboxState

### DISCUSSION THREAD

**Discussion by Bravemobin** [igordanchenko/yet-another-react-lightbox#128] [Q&A] [answered]
when I try to use useLightboxState() next js give me this error

> "Error: useLightboxState must be used within a LightboxStateContext.Provider"

and next I use <LightboxStateContextProvider> give me this error 

> "TypeError: Cannot read properties of undefined (reading 'length')"

I have no idea what should I do

**Comment by igordanchenko** [comment]
Lightbox hooks can be used only in a component rendered inside the lightbox (i.e., custom button, or any other component you add to the lightbox with a plugin).

if you want to access the lightbox state outside the lightbox, you can use the [Controller ref](https://yet-another-react-lightbox.com/documentation#ControllerRef).

Can you share your draft, and describe what are you trying to implement?

**Reply by Bravemobin** [reply]
thanks I going to check this In document
I want use currentindex to access my arrays element 
and another question, does lightbox support add items to array onclick next item or previous item ?
example I have 50 image I want when currentindex in 45 I load more images

**Reply by igordanchenko** [reply, marked as answer]
You probably want to use the `on.view` callback in this case.

```jsx
<Lightbox
  on={{ 
    view: ({ index }) => { 
      // ...
    } 
  }}
/>
```

Here is an example demonstrating how you can dynamically add slides to the lightbox - https://codesandbox.io/p/sandbox/yet-another-react-lightbox-128-7mczyy?file=%2Fsrc%2FApp.tsx

**Reply by igordanchenko** [reply]
@Bravemobin, does this answer your question?

**Reply by Alireza-Ghavabesh** [reply]
hi
i have similar question, i added more props to my slides like this:

```jsx
slides = [
    {
       src: "some url",
       user_id: user_id,
       image_id: image_id,
    }
]
```

how can i access to this custom props (user_id and image_id) ?
i added button with toolbar and i want access to this props when clicked button.

**Reply by Alireza-Ghavabesh** [reply]
i found solution:

```jsx
on={{
  view: () => console.log(ref.current.getLightboxState().currentSlide),
}}
```
